README.md

```diff
@@ -20,110 +13,1177 @@ visit https://pkg.go.dev/github.com/pinecone-io/go-pinecone/pinecone.
+	if err != nil {
+		log.Fatalf("Failed to describe index \"%s\". Error:%s", idx.Name, err)
+	} else {
+		fmt.Printf("Successfully found the \"%s\" index!\n", idx.Name)
+	}
+}
+```
 
-The API Definitions are in a private submodule. To checkout or update the submodules execute in the root of the project:
+## Index Operations
 
-```shell
-git submodule update --init --recursive
+Pinecone indexes support working with vector data using operations such as upsert, query, fetch, and delete.
+
+### Targeting an index
+
+To perform data operations on an index, you target it using the `Index` method on a `Client` object. You will
+need your index's `Host` value, which you can retrieve via `DescribeIndex` or `ListIndexes`.
+
+```go
+package main
+
+import (
+	"github.com/pinecone-io/go-pinecone/pinecone"
+	"os"
+)
+
+func main() {
+    ctx := context.Background()
+
+    pc, err := pinecone.NewClient(pinecone.NewClientParams{
+        ApiKey: "YOUR_API_KEY",
+    })
+    if err != nil {
+        log.Fatalf("Failed to create Client: %v", err)
+    }
+
+    idx, err := pc.DescribeIndex(ctx, "pinecone-index")
+    if err != nil {
+        log.Fatalf("Failed to describe index \"%v\": %v", idx.Name, err)
+    }
+
+    idxConnection, err := pc.Index(pinecone.NewIndexConnParams{Host: idx.Host})
+    if err != nil {
+        log.Fatalf("Failed to create IndexConnection for Host: %v: %v", idx.Host, err)
+	}
+}
+```
+
+### Upsert vectors
+
+The following example upserts vectors to `example-index`.
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+
+    "github.com/pinecone-io/go-pinecone/pinecone"
+)
+
+func main() {
+    ctx := context.Background()
+
+    pc, err := pinecone.NewClient(pinecone.NewClientParams{
+        ApiKey: "YOUR_API_KEY",
+    })
+    if err != nil {
+        log.Fatalf("Failed to create Client: %v", err)
+    }
+
+    idx, err := pc.DescribeIndex(ctx, "example-index")
+    if err != nil {
+        log.Fatalf("Failed to describe index \"%v\": %v", idx.Name, err)
+    }
+
+    idxConnection, err := pc.Index(pinecone.NewIndexConnParams{Host: idx.Host})
+    if err != nil {
+        log.Fatalf("Failed to create IndexConnection for Host: %v: %v", idx.Host, err)
+	}
+
+    vectors := []*pinecone.Vector{
+        {
+            Id:     "A",
+            Values: []float32{0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1},
+        },
+        {
+            Id:     "B",
+            Values: []float32{0.2, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2},
+        },
+        {
+            Id:     "C",
+            Values: []float32{0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3},
+        },
+        {
+            Id:     "D",
+            Values: []float32{0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4},
+        },
+    }
+
+    count, err := idxConnection.UpsertVectors(ctx, vectors)
+    if err != nil {
+        log.Fatalf("Failed to upsert vectors: %v", err)
+    } else {
+        log.Fatalf("Successfully upserted %d vector(s)", count)
```

Comment:
the `else` stmts in these examples should not have `log.Fatalf`. Should be `fmt.Printf` (+ have "fmt" as an import)